node/router/src/lib.rs

```diff
@@ -422,15 +516,19 @@ pub trait PeerPoolHandling<N: Network>: P2P {
     }
 
     /// Temporarily IP-ban and disconnect from the peer with the given listener address and an
-    /// optional reason for the ban.
+    /// optional reason for the ban. This also removes the peer from the candidate pool.
     fn ip_ban_peer(&self, listener_addr: SocketAddr, reason: Option<&str>) {
         let ip = listener_addr.ip();
         debug!("IP-banning {ip}{}", reason.map(|r| format!(" reason: {r}")).unwrap_or_default());
 
+        // Insert/update the low-level IP ban list.
         let tcp = self.tcp().clone();
```

Comment:
Is this clone needed?

Reply:
It is in fact superfluous; I don't know why it was there originally, perhaps a remnant of back when it was delegated to some spawned task.